spidermon/contrib/monitors/mixins/validation.py

```diff
@@ -84,125 +84,259 @@ def __init__(self, stats, prefix=None):
+            if missing_count > allowed_count:
+                msg = self._get_msg_for_missing_required_count(field_name, missing_count, allowed_count)
+                msgs.append(msg)
+        if msgs:
+            msgs.insert(0, 'Required fields are missing:')
+            self.fail('\n'.join(msgs))
 
     def check_missing_required_field(self, field_name, allowed_count=0):
+        """
+        Checks that the number of "missing_required_field" errors for the ``field_name`` field is less or equal than
+        ``allowed_count``.
+        """
         missing_count = (
             self.validation.fields[field_name].errors["missing_required_field"].count
         )
-        self._check_missing_required_count(missing_count, allowed_count)
-
-    def _check_missing_required_count(self, missing_count, allowed_count):
-        self.assertLessEqual(
-            missing_count,
-            allowed_count,
-            msg="{count} required field{plural} are missing!{threshold_info}".format(
-                count=missing_count,
-                plural="" if missing_count == 1 else "s",
-                threshold_info=(" (maximum allowed %d)" % allowed_count)
-                if allowed_count > 0
-                else "",
-            ),
+        msg = self._get_msg_for_missing_required_count(field_name, missing_count, allowed_count)
+        self.assertLessEqual(missing_count, allowed_count, msg)
+
+    @staticmethod
+    def _get_msg_for_missing_required_count(field_name, missing_count, allowed_count):
+        msg = "Required field {field} is missing in {count} items!{threshold_info}".format(
+            count=missing_count,
+            field=field_name,
+            threshold_info=(" (maximum allowed %d)" % allowed_count)
+            if allowed_count > 0
+            else "",
         )
+        return msg
 
     def check_missing_required_fields_percent(
         self, field_names=None, allowed_percent=0
     ):
-        if not field_names:
+        """
+        Checks that the number of "missing_required_field" errors for the ``field_names`` fields divided by the number
+        of items is less or equal than ``allowed_percent`` and raises an error with all problematic fields.
+        If ``field_names`` is None and ``self.new_behavior`` is True, checks all fields.
+        If ``field_names`` is None and ``self.new_behavior`` is False, checks that the total number of
+        "missing_required_field" errors is less or equal than ``allowed_count``.
+        """
+        if not self.new_behavior and not field_names:
+            # TODO: deprecate this
             missing_percent = self.validation.errors["missing_required_field"].percent
-            self._check_missing_required_percent(missing_percent, allowed_percent)
-        else:
-            for field_name in field_names:
-                self.check_missing_required_field_percent(field_name, allowed_percent)
+            self.assertLessEqual(
```

Comment:
Here is an assertion for _less or equal_, but a couple of lines below, it is for _greater than_.
The behavior is the same, but keeping only one conditional would make it easier to understand.

Reply:
The assert line means "it's good if it's <=" while the manual comparison line means "it's bad if it's >", so just reversing the manual comparison to read `if allowed_percent < missing_percent` looks unhelpful to me, should it read `if not (missing_percent <= allowed_percent)`?

Reply:
It is more of an idea.
If you think it is understandable as is, it is fine.
My concern was only to keep the statements similar.